src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -114,10 +115,13 @@ public JobBuilder newJob(
 
   // submit a new job to stairway
   // protected method intended to be called only from JobBuilder
-  protected String submit(
-      Class<? extends Flight> flightClass, FlightMap parameterMap, String jobId) {
+  protected String submit(Class<? extends Flight> flightClass, FlightMap parameterMap, String jobId)
+      throws DuplicateFlightIdSubmittedException {
     try {
       stairway.submit(jobId, flightClass, parameterMap);
+    } catch (DuplicateFlightIdSubmittedException ex) {
+      // This needs to be thrown separately as it extends the more general StairwayException
+      throw ex;
     } catch (StairwayException | InterruptedException stairwayEx) {
```

Comment:
Probably not this PR, but it smells that JobService throws unchecked exceptions for some but not all of Stairway. It's leaking the Stairway API significantly, but still wrapping it. JobService should probably be pushed one way or another - more wrapping or less.

Reply:
For the most part I think we should generally be wrapping Stairway (`InternalStairwayException` is a WM type) -  I'll change `DuplicateJobIdSubmittedException` to stay inside JobService.